astropy/units/equivalencies.py

```diff
@@ -171,7 +188,7 @@ def spectral():
 @deprecated_renamed_argument(
     "factor", None, since="7.0", alternative='"wav" as a "Quantity"'
 )
-def spectral_density(wav, factor=None):
+def spectral_density(wav: Quantity, factor: None = None) -> Equivalency:
```

Comment:
```suggestion
def spectral_density(wav: Quantity, factor: ArrayLike | None = None) -> Equivalency:
```

?

Reply:
`factor` is deprecated. I don't want to spend effort on annotating a feature that shouldn't be used because it will be removed.

Reply:
Is the effort not pressing "accept" on the suggestion?

Reply:
If `factor` is not `None` then `wav` must be a `UnitBase`, not `Quantity`, and that means we would have to use overloads just so that some hypothetical user (that might not even exist) could get a runtime deprecation warning instead of a notification from their IDE.

Reply:
Ok. Let's skip it. We can chalk it up to part of the deprecation process that mypy will complain! 🤣

Reply:
Mypy has no complaints about this by default.

Reply:
About passing still-valid (if deprecated) non-None values?